ethpm/backends/ipfs.py

```diff
@@ -0,0 +1,123 @@
+import requests
+
+from abc import abstractmethod
+
+from eth_utils import to_bytes
+
+from ethpm import V2_PACKAGES_DIR
+
+from ethpm.constants import (
+    INFURA_GATEWAY_PREFIX,
+    IPFS_GATEWAY_PREFIX,
+)
+
+from ethpm.backends.base import BaseURIBackend
+
+from ethpm.utils.ipfs import (
+    extract_ipfs_path_from_uri,
+    is_ipfs_uri,
+)
+
+
+class BaseIPFSBackend(BaseURIBackend):
+    """
+    Base class for all URIs with an IPFS scheme.
+    """
+    @abstractmethod
+    def fetch_uri_contents(self, uri: str) -> bytes:
+        """
+        Return the file contents stored at the URI.
+        """
+        raise NotImplementedError("Must be implmented by subclass.")
+
+    def can_handle_uri(self, uri: str) -> bool:
+        """
+        Return a bool indicating whether or not this backend
+        is capable of serving the content at the URI.
+        """
+        if not is_ipfs_uri(uri):
+            return False
+        return True
+
+    @property
+    @abstractmethod
+    def base_uri(self) -> str:
+        """
+        Return the base URI unique to a particular scheme.
+        """
+        raise NotImplementedError("Must be implemented by subclass.")
+
+
+class IPFSOverHTTPBackend(BaseIPFSBackend):
+    """
+    Base class for all IPFS URIs served over an http connection.
+    """
+    @property
+    @abstractmethod
+    def base_uri(self) -> str:
```

Comment:
No need to repeat the abstract method here as it is already in the `BaseIPFSBackend` just leave it out and it will implicitly continue to be the abstract method from the base class preventing anyone from direct instantiation of `IPFSOverHTTPBackend`